### ISSUE TITLE
Use primary identifiers as device registry keys

### ISSUE THREAD

**Issue by johanstokking** [TheThingsNetwork/lorawan-stack#260]
**Summary:** 

Use primary identifiers as keys in Join Server while keeping EUIs unique.

**Why do we need this?**

To do device management with primary identifiers, but lookup of join-request by EUIs.

**What is already there? What do you see now?**

- Database key is EUI
- Device registry needs the EUIs as part of the given end device identifiers, but in a `GET` these are not available through the gRPC gateway

**What is missing? What do you want to see?**

- Use primary identifiers as database key
- Use EUI key pointing to the primary identifiers

Basically the same functionality as in NS.

**How do you propose to implement this?**

Copy NS functionality.

**What can you do yourself and what do you need help with?**

Will do.

**Comment by johanstokking** [comment]
Will be merged with #194
